dali/python/nvidia/dali/pipeline.py

```diff
@@ -478,6 +478,12 @@ def is_restored_from_checkpoint(self):
         """If True, this pipeline was restored from checkpoint."""
         return self._is_restored_from_checkpoint
 
+    @property
+    def num_outputs(self):
```

Comment:
This is a new public API function and, as such, it should be robust - now it doesn't work for deserialized pipelines (they don't populate `_graph_outputs` nor `_names_and_devices`. It calls for fixing, but not in this PR. My suggestion would be to make this property temporarily private.